gcf/custom/manifest.go

```diff
@@ -25,6 +26,40 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+func isAlphaNumeric(c byte) bool {
+	return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9')
+}
+
+func computeImportGroup(root string) (string, error) {
+	importGroup := path.Base(root)
+	if len(importGroup) == 0 {
+		return "", fmt.Errorf("empty importGroup: %s", root)
+	}
+
+	// IMPORTANT NOTE:
+	// importGroup has a character constraint of 20 due to internal limitations.
+	// https://cloud.google.com/storage-transfer/docs/known-limitations-transfer#tsop-max-path
+	if len(importGroup) > 20 {
+		importGroup = importGroup[:20]
+	}
+
+	// Strip any trailing non-alphanumeric characters.
+	i := len(importGroup)
+	for !isAlphaNumeric(importGroup[i-1]) {
+		i--
+		if i == 0 {
+			break
+		}
+	}
+	importGroup = importGroup[:i]
```

Comment:
why not do the trimming first then take the first 20?

Reply:
This is to handle cases where the truncated name ends with symbols 
(e.g. "sustainable_systems_lab" --> "sustainable_systems_" --> "sustainable_systems") 

Alternatively I could just remove all non-alphanumeric characters from the name

Wdyt?

Reply:
Yea, that sg and leaves more room for truncation!

Reply:
oh, do you want to make the change to remove non-alphanum chars?

Reply:
Done